app/scripts/controllers/push-platform-notifications/services/services.ts

```diff
@@ -195,18 +195,21 @@ export async function listenToPushNotifications(
   const unsubscribePushNotifications = onBackgroundMessage(
     messaging,
     async (payload: MessagePayload): Promise<void> => {
-      const typedPayload = payload;
-
-      // if the payload does not contain data, do nothing
       try {
-        const notificationData: NotificationUnion = typedPayload?.data?.data
-          ? JSON.parse(typedPayload?.data?.data)
+        const data = payload?.data?.data
+          ? JSON.parse(payload?.data?.data)
           : undefined;
 
-        if (!notificationData) {
+        // if the payload does not contain data, do nothing
+        if (!data) {
           return;
         }
 
+        const notificationData = {
+          ...data,
+          type: data?.type ?? data?.data?.kind,
+        } as NotificationUnion;
```

Comment:
HERE - not all push notification data shapes have a type we can discriminate over, hence this needs to be polyfilled with the correct information.

As soon as we do give the correct type, we can then process the notification correctly and show the correct push notification message.

Reply:
@MetaMask/notifications lets discuss making the push notifications uniform (especially once we support other notification types/groups.)